### PR TITLE
Make it easier to reuse kubelet server code

### PR DESCRIPTION
**Pull request by smarterclayton**
Ensure that RunKubelet() returns errors consistently, and make info output
match Kube conventions.

Allows OpenShift to more easily reuse the Kubelet.

Also sets FakeDocker to emulate Docker 1.6, and ensures DefaultHealthz can't
stomp on itself.

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func (s *KubeletServer) Run(_ []string) error {
 
 	// TODO(vmarmol): Do this through container config.
 	if err := util.ApplyOomScoreAdj(0, s.OOMScoreAdj); err != nil {
-		glog.Info(err)
+		glog.V(4).Info(err)
```
Comment (vmarmol) [inline cmd/kubelet/app/server.go]: This is printed once at startup, is there a particular reason to make it such a low V value? Knowing if this triggered helps a lot under OOM conditions.

Comment (smarterclayton) [inline cmd/kubelet/app/server.go]: I was just applying based sweep.  I can put it back to regular.

Comment (smarterclayton) [inline cmd/kubelet/app/server.go]: Any reason it's not a Warning?

Comment (smarterclayton) [inline cmd/kubelet/app/server.go]: Turned it into a Warning

----- Original Message -----

Comment (vmarmol) [inline cmd/kubelet/app/server.go]: Thanks, yeah it should be.

**Comment by bgrant0607** [comment]
@smarterclayton Openshift defines its own kubelet command? What does it need to do differently?

**Comment by vmarmol** [comment]
LGTM, will merge on green.

**Comment by smarterclayton** [comment]
@bgrant0607 we simply launch the Kubelet with all our security settings preapplied (as part of a unified binary), but originally (pre hyperkube) we couldn't reuse any code.  This was some cleanup I hit while trying to make RunKubelet fit a bit more into the style of being called from external code.

**Pull request merged**